### PR TITLE
Migrate BaseCommand derived commands to argparse

### PR DESCRIPTION
**Pull request by dwaynebailey**
This focuses on all commands that inherit directly from BaseCommand

### CODE REVIEW

**Comment by dwaynebailey** [comment]
@translate/core this is ready for review.

- I've kept all command changes in one commit
- Bugs found through testing are next each on their own
- Lastly, each command is tested.  I will likely squash these test commits once reviewed.

File: tests/commands/webpack.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from django.core.management import call_command
+
+
+@pytest.mark.cmd
+def test_webpack(capfd):
```
Comment (julen) [inline tests/commands/webpack.py]: In its current shape, this test doesn't add any value, as it's not asserting anything but just running the command.

**Comment by julen** [comment]
The code changes look good to me overall.

File: tests/commands/webpack.py (hunk 1 above)
Comment (dwaynebailey) [inline tests/commands/webpack.py]: @julen yes wanted to ping you on this one.  Wasn't too sure how best to test this or if we should test it.

File: tests/commands/initdb.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@pytest.mark.django_db
+def test_initdb_noprojects(capfd):
+    """Initialise the database with initdb
+
+    Testing without --no-projects would take too long
+    """
+    call_command('initdb', '--no-projects')
+    out, err = capfd.readouterr()
+    assert "Successfully populated the database." in out
+    assert "Successfully populated the database." in out
```
Comment (unho) [inline tests/commands/initdb.py]: Repeated assert.

File: tests/commands/test_checks.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks', '--unit=100000')
+    assert 'Unit matching query does not exist' in str(e)
+
+
+@pytest.mark.cmd
+@pytest.mark.django_db
+def test_test_checks_unit(capfd, site_matrix):
+    """Check a --unit"""
+    # Unit 3/104 is /language0/project0/store0.po Translated Target
```
Comment (unho) [inline tests/commands/test_checks.py]: If this comment refers to 3 and 196 below, the numbers don't match.

File: tests/commands/test_checks.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+
+@pytest.mark.cmd
+def test_test_checks_srctgt_missing_args():
+    """Check a --source --target with incomplete options."""
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks', '--source="%s files')
+    assert 'Use a pair of --source and --target' in str(e)
+    with pytest.raises(CommandError) as e:
+        call_command('test_checks', '--target="%s leers')
```
Comment (unho) [inline tests/commands/test_checks.py]: I think that using the `%s` placeholder in the test string might confuse anyone reading the code. I suggest using just `Files` and `Leers`.

File: tests/commands/import.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+msgstr ""
+"X-Pootle-Path: /af/tutorial/tutorial.po\\n"
+"X-Pootle-Revision: 12\\n"
+
+msgid "rest"
+msgstr "test"
+           """)
+    call_command('import', os.path.join(p.dirname, p.basename))
+    out, err = capfd.readouterr()
+    assert "Updating /af/tutorial/tutorial.po" in err
```
Comment (unho) [inline tests/commands/import.py]: Shouldn't this be in `out`?

File: tests/commands/import.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+msgstr ""
+"X-Pootle-Path: /af/tutorial/tutorial.po\\n"
+"X-Pootle-Revision: 12\\n"
+
+msgid "rest"
+msgstr "test"
+           """)
+    call_command('import', '--user=member', os.path.join(p.dirname, p.basename))
+    out, err = capfd.readouterr()
+    assert "Updating /af/tutorial/tutorial.po" in err
```
Comment (unho) [inline tests/commands/import.py]: Same as in test above.

**Comment by unho** [comment]
Reviewed. lgtm besides Travis failures and a bunch of tests just running commands. Also the tests could be a bit more extensive, but perhaps it is not worth the effort.

File: tests/commands/import.py (hunk 5 above)
Comment (dwaynebailey) [inline tests/commands/import.py]: Does the test fail?

Comment (unho) [inline tests/commands/import.py]: This was one of the three tests that I saw that failed on Travis.

File: tests/commands/webpack.py (hunk 1 above)
Comment (dwaynebailey) [inline tests/commands/webpack.py]: I ended up just droping the webpack test.

**Pull request merged**